pkg/cmd/util/clientcmd/factory.go

```diff
@@ -201,7 +201,11 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			if !ok {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
-			buildsForBCSelector := labels.SelectorFromSet(map[string]string{buildapi.DeprecatedBuildConfigLabel: t.Name})
+			buildConfigLabels := labels.Set{
+				buildapi.BuildConfigLabel: t.Name,
```

Comment:
you should probably do what @kargakis suggested and get everything and then filter it client side, rather than running two queries.

(the current logic is already getting everything and filtering it anyway, as i understand it, selectors are applied client side)